pkg/storegateway/bucket.go

```diff
@@ -1798,10 +1730,12 @@ func newBucketBlockSet() *bucketBlockSet {
 	return &bucketBlockSet{}
 }
 
-func (s *bucketBlockSet) add(b *bucketBlock) error {
+func (s *bucketBlockSet) add(b *bucketBlock) {
 	s.mtx.Lock()
```

Comment:
It looks like we always acquire the mutex before making changes to `s.blockSet` so that it remains consistent with `s.blocks`, even though `s.blockSet` is internally synchronized. Can we document that somewhere? In each method were we acquire the mutex or next to its definition?